main/vlan/mvlan.pre-up

```diff
@@ -22,9 +27,7 @@ if ! ip link show "$RAW_DEVICE" >/dev/null; then
 	exit 1
 fi
 
-ip link add link "$RAW_DEVICE" name "$IFACE" type macvlan
-
-if ! ip link show "$IFACE" >/dev/null; then
-	echo "Failed to create vlan device $IFACE on device $RAW_DEVICE with tag $VLAN_ID"
-	exit 1
+trap "device_creation_error" ERR
```

Comment:
Why do you use `trap` here? Why not just add `|| device_creation_error` after execution of `ip` command?

Reply:
I used trap to get something similar to the way I did in `vlan.pre-up`. If you prefer, I could add `|| device_creation_error` instead.